src/main/java/Duke.java

```diff
@@ -1,10 +1,603 @@
+
+        if (data.exists()) {
+            showGreetings();
+            acceptUserInputs(data);
+            showGoodbye();
+        }
+    }
+
+    /***
+     * Reads in the user inputs and processes the commands.
+     */
+    private static void acceptUserInputs(File data) {
+        try {
+            readExistingData(data, storedValues);
+        } catch (FileNotFoundException e) {
+            System.out.println("File is not found!");
+        }
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();
+        while (!hasProcessedAllInputs(line, storedValues)) {
+            line = in.nextLine();
+        }
+    }
+
+    /***
+     * Reads the existing data within the txt file before new user inputs are allowed. Ensures that the
+     * list is the most updated since the last interaction with Duke.
+     * @param data The txt file in the local environment.
+     * @param storedValues List of inputs stored by the user.
+     * @throws FileNotFoundException Thrown when file is not found in the environment.
+     */
+    private static void readExistingData(File data, ArrayList<Task> storedValues) throws FileNotFoundException {
+        Scanner readData = new Scanner(data);
+        while (readData.hasNext()) {
+            // Previous data stored from calling duke
+            String extractType[] = readData.nextLine().split(" | ", 5);
+            String type = extractType[0];
+            String marked = extractType[2];
+            String task = extractType[4];
+
+            switch (type) {
+            case "T":
+                readExistingToDo(storedValues, marked, task);
+                break;
+            case "D":
+                readExistingDeadline(storedValues, extractType, marked);
+                break;
+            case "E":
+                readExistingEvent(storedValues, extractType, marked);
+                break;
+            }
+        }
+    }
+
+    /***
+     * Reads the existing events within the txt file, processes it and displays it when the command "list"
+     * is called by the user.
+     * @param storedValues List of inputs stored by the user.
+     * @param extractType Description of task with duration.
+     * @param marked Status of marked in txt file.
+     */
+    private static void readExistingEvent(ArrayList<Task> storedValues, String[] extractType, String marked) {
+        int firstSlash = extractType[4].indexOf("|");
+        String eventName = extractType[4].substring(0,firstSlash-1);
+        String eventDuration = extractType[4].substring(firstSlash+2);
+        int secondSlash = eventDuration.indexOf("|");
+        String eventFrom = eventDuration.substring(0,secondSlash-1);
+        String eventTo = eventDuration.substring(secondSlash+2);
+        Event addInputE = new Event(eventName, eventFrom, eventTo);
+        storedValues.add(taskNum,addInputE);
+        if (marked.equals("1")) {
+            storedValues.get(taskNum).markAsDone();
+        }
```

Comment:
Some repeated code here with readExistingDeadline() method. Perhaps think if it is possible to write smaller methods and reuse them?